File: Data Structures & Algorithms/sort-an-array/submission-1.py
import heapq

class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        heap = []
        heapq.heapify(heap)

        for num in nums:
            heapq.heappush(heap, num)

        result = []
        while len(heap) > 0:
            item = heapq.heappop(heap)
            result.append(item)

        return result

        

File: Data Structures & Algorithms/subsets/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    subsets(nums) {
        const newArray = [[]]
        for ( const num of nums) {
            const newValues = []

            for (const item of newArray) {
                newValues.push([...item, num])
            }

            newArray.push(...newValues)
        }

        return newArray
    }
}
